### DISCUSSION TITLE
how to work with innerRef  with array

### DISCUSSION THREAD

**Discussion by CurryHan** [react-hook-form/react-hook-form#6951] [Q&A]
const firstName = register('firstName', { required: true })
<TextInput 
  onChange={firstName.onChange}
  onBlur={firstName.onBlur}
  inputRef={firstName.ref} // you can achieve the same for different ref name such as innerRef
/>

how to register array

**Comment by bluebill1049** [comment]
> how to register array

what do you mean by register array?

**Reply by CurryHan** [reply]
I need to register like 'array.0.keyvalue',but how I define like `const firstName = register('firstName', { required: true })`

**Reply by bluebill1049** [reply]
`const firstName = register('array.0.keyvalue', { required: true })` ?

**Reply by CurryHan** [reply]
> `const firstName = register('array.0.keyvalue', { required: true })` ?

Thank your quick reply.
if the array length is 3,how to register all?

**Reply by bluebill1049** [reply]
`[0.1,2].forEach(() => register(...))`?

**Reply by CurryHan** [reply]
> () => register(...

if to do like this,can I get the defined const variable?

**Reply by bluebill1049** [reply]
what do you mean by const variable?

**Reply by CurryHan** [reply]
I'm sorry,my english is not good
I have a custom select component,and the select component is in the table,
```
<table>
   <tr>
      <td>
             <CustomSelect  {...register("array.0.key")}></CustomSelect>
     </td>
</tr>
<tr>
      <td>
             <CustomSelect  {...register("array.1.key")}></CustomSelect>
     </td>
</tr>
   </columns>
</Table>
```
if  I change the select ,it show me the error below

![image](https://user-images.githubusercontent.com/14214176/140022444-986f1110-ed7a-4ad2-9c52-4385e82c9dd7.png)

**Reply by bluebill1049** [reply]
That's all good, why not just wrap it with Controller? https://react-hook-form.com/api/usecontroller/controller it's probably the right component for the work.

**Reply by CurryHan** [reply]
thanks,I tried
step 1,I append array of Length 2
step 2,I remove the array
step 3,I append array of Length 3
step 4, I submit,but the values length is 5
```
//step 1
var array=[0,1]
append(array)
//step 2
const [tempList, setTempList] = useState([])
const change=()=>{
   remove()
   //step 3
   setTempList([0,1,2])
}
useEffect(() => {
        append(tempList)
    }, [tempList, append])

//step 3
const submit = async (values) => {
        console.log(values)//values length is 5
    }
```
